lib/hooks.dart

```diff
@@ -0,0 +1,97 @@
+@JS()
+library hooks;
```

Comment:
After going back and forth on it, I suggested this offline to makes things a little more organized as opposed to just lumping more things into `react.dart`. Also, since they're APIs that will only be used sometimes, I figured it made sense for them to be their own self-contained entrypoint.

But, I could be convinced to move them into `react.dart`.

Reply:
I feel the same it makes sense to have them seperate but i also enjoy the convenience of getting them when importing `react.dart`... plus they would get tree shaken out if they weren't used right?